src/main/kotlin/pl/allegro/mobile/logic/operators/string/CompareToDateOperation.kt

```diff
@@ -0,0 +1,32 @@
+package pl.allegro.mobile.logic.operators.string
+
+import pl.allegro.mobile.logic.ClientLogicElement
+import pl.allegro.mobile.logic.ClientLogicMarker
+import pl.allegro.mobile.logic.ClientLogicOperator
+import pl.allegro.mobile.logic.StringElement
+
+internal interface  CompareToDateOperation {
+    /**
+     * Compares a string to given string representing the date in a proper format.
+     * This operation compares the date
+     * @receiver Character sequence or client side operation that returns string
+     * @param date date to compare to date represents in current string
+     * @return integer interpretation of state. State is between -1 and 1.
+     * -1 - passed date is AFTER the compared date
+     * 0 - passed date is EQUALS the compared date
+     * 1 - passed date is BEFORE the compared date
+     */
+    @ClientLogicMarker
+    fun ClientLogicElement.compareToDate(date: String) =
+        CompareToDateOperationFactory().create(this,date)
+}
+
+private class CompareToDateOperationFactory {
```

Comment:
can you also add an option to compare two dates stored in the registry?

Reply:
👍🏻 It should be looks like this?

Reply:
+1, but the parameter name should be more generic, e.g. `element` because you can pass any operation result there. For evemple you can also create following expression:

`registryKey("date").compareToDate(concat("%s-01-01", registryKey("year") ))`
It'd also be good to have such example in tests